core/ui/src/main/kotlin/nl/q42/template/ui/theme/Theme.kt

```diff
@@ -23,64 +21,62 @@ import androidx.core.view.WindowCompat
  * README file of our project.
  */
 
-private val DarkColorScheme = darkColorScheme(
-    primary = Purple80,
-    secondary = PurpleGrey80,
-    tertiary = Pink80,
-    background = Black,
-)
-
-private val LightColorScheme = lightColorScheme(
-    primary = Purple40,
-    secondary = PurpleGrey40,
-    tertiary = Pink40,
-    background = White
-
-    /* Other default colors to override
-    background = Color(0xFFFFFBFE),
-    surface = Color(0xFFFFFBFE),
-    onPrimary = Color.White,
-    onSecondary = Color.White,
-    onTertiary = Color.White,
-    onBackground = Color(0xFF1C1B1F),
-    onSurface = Color(0xFF1C1B1F),
-    */
-)
+private val LocalAppTypography = staticCompositionLocalOf { AppTypography() }
+private val LocalAppColorTokens = staticCompositionLocalOf<AppColorTokens> {
+    // Dummy default, will be replaced for the actual tokens by the Provider
+    AppColorTokensLight
+}
+private val LocalAppShapes = staticCompositionLocalOf { AppShapes() }
 
 @Composable
 fun AppTheme(
     darkTheme: Boolean = isSystemInDarkTheme(),
-    // Dynamic color is available on Android 12+
-    dynamicColor: Boolean = true,
+    typography: AppTypography = AppTheme.typography,
+    colors: AppColorTokens = AppTheme.colors,
+    shapes: AppShapes = AppTheme.shapes,
+
     content: @Composable () -> Unit
 ) {
-    val colorScheme = when {
-        dynamicColor && Build.VERSION.SDK_INT >= Build.VERSION_CODES.S -> {
-            val context = LocalContext.current
-            if (darkTheme) dynamicDarkColorScheme(context) else dynamicLightColorScheme(context)
-        }
-
-        darkTheme -> DarkColorScheme
-        else -> LightColorScheme
-    }
     val view = LocalView.current
     if (!view.isInEditMode) {
         SideEffect {
             val window = (view.context as Activity).window
-            window.statusBarColor = colorScheme.primary.toArgb()
+            window.statusBarColor = colors.accent.toArgb()
             WindowCompat.getInsetsController(window, view).isAppearanceLightStatusBars = darkTheme
         }
     }
 
-    MaterialTheme(
```

Comment:
We still need the MaterialTheme wrapper for system Composables like AlertDialog

Reply:
Nah, we can leave it out. When adding a dialog, just explicitly set App colors in there